core/src/mender-artifact-download.c

```diff
@@ -53,6 +53,12 @@ mender_download_artifact(const char *uri, mender_deployment_data_t *deployment_d
         return ret;
     }
 
+    /* If the artifact doesn't contain a file, the update module won't be set */
```

Comment:
Discussed face to face,

An Artifact with no files is a valid use case of an "update" and the MCU Client should handle that.

Without looking at the code, my gut feeling is that setting we are setting `dl_data.update_module` too late (when parsing files) while we should do it as soon as we knot the artifact type